core/system.h

```diff
@@ -74,13 +89,26 @@ class Device
 
     void request_autopilot_version();
 
-    uint64_t get_target_uuid() const;
-    uint8_t get_target_system_id() const;
-    uint8_t get_target_component_id() const;
+    // Adds unique component ids
+    void add_new_component(uint8_t component_id);
+    size_t total_components() const;
+
+    uint8_t get_autopilot_id() const;
+    std::vector<uint8_t> get_camera_ids() const;
+    uint8_t get_gimbal_id() const;
+
+    bool is_standalone() const;
+    bool has_autopilot() const;
+    bool has_camera() const;
```

Comment:
As discussed, there might be multiple cameras. This API not sufficient.

Reply:
Ok! I shall change it. Thanks.